tests/main/smoke/task.yaml

```diff
@@ -0,0 +1,99 @@
+summary: smoke test
+
+prepare: |
+  # for various utilities
+  . "$TESTSLIB/nested.sh"
+
+  # Start the nested UC VM
+  start_nested_core_vm_unit
+
+
+execute: |
+  # for various utilities
+  . "$TESTSLIB/nested.sh"
+
+  # expect prompt from console conf on serial
+  wait_for_serial_prompt "$(get_nested_svc_name)" "$WORK_DIR" "Press enter to configure."
+  # wait for console conf setup to complete
+  wait_for_serial_prompt "$(get_nested_svc_name)" "$WORK_DIR" "console-conf tweaks done"
+
+  # the VM may be very slow
+  sleep 30
+
+  mkdir -p screenshots
+
+  nested_vm_screenshot screenshots/01-after-boot.png
+
+  nested_vm_send_raw_input "kp_enter"
+  sleep 30
+
+  # transitioning from the 'Press enter to configure.' screen to actual welcome
+  # displayed by console-conf can take quite a while, try a couple of times.
+  cnt=10
+  while true; do
+      nested_vm_screenshot screenshots/02-welcome.png
+      if ! "$TESTSLIB/tools/imcmp.py" screenshots/02-welcome.png welcome-reference.png \
+              --mask welcome-mask.png; then
+          echo "awaiting output to be similar"
+
+          cnt=$((cnt-1))
+          if [ "$cnt" -eq 0 ]; then
+              echo "timeout waiting for reference screen to appear"
+              exit 1
+          fi
+          sleep 5
+      else
+          # we are at the welcome screen
+          break
+      fi
+  done
+
+  nested_vm_send_raw_input "kp_enter"
+  sleep 30
+
+  nested_vm_screenshot screenshots/03-network-select.png
```

Comment:
I see that some screenshots are taken but not compared, which I think is fine, but maybe it is worth adding a TODO to check also those screens in the future.

Also, I think these are copied to artifacts, is that correct?

Reply:
yes, artifacts are always collected whether the task succeeds or fails, the artifact-upload step runs always and  will make them available for inspection even if the test failed